src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Windows.cs

```diff
@@ -6,9 +6,25 @@ namespace System.Globalization
 {
     internal static partial class GlobalizationMode
     {
-        private static bool GetGlobalizationInvariantMode()
+        // Order of these properties in Windows matter because GetUseIcuMode is dependent on Invariant.
+        // So we need Invariant to be initialized first.
+        internal static bool Invariant { get; } = GetInvariantSwitchValue();
+
+        internal static bool UseNls { get; } = GetUseNlsMode();
+
+        private static bool GetUseNlsMode()
         {
-            return GetInvariantSwitchValue();
+            bool retVal = true;
+            if (!Invariant)
+            {
+                if (!GetSwitchValue("System.Globalization.UseNls", "DOTNET_SYSTEM_GLOBALIZATION_USENLS") &&
+                                    Interop.Globalization.LoadICU() != 0)
+                {
+                    retVal = false;
+                }
+            }
+
+            return retVal;
```

Comment:
```suggestion
            if (Invariant)
                return false;
                
            return GetSwitchValue("System.Globalization.UseNls", "DOTNET_SYSTEM_GLOBALIZATION_USENLS") ||
                                    Interop.Globalization.LoadICU() == 0;
```

Reply:
Good suggestion. Thanks.

Reply:
I think we can remove the intermediate `GetUseNlsMode` method and do:
```cs
internal static bool UseNls { get; } = !Invariant && (GetSwitchValue("System.Globalization.UseNls", "DOTNET_SYSTEM_GLOBALIZATION_USENLS") ||
                                         Interop.Globalization.LoadICU() == 0);
```

Will instead do that.